core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/organism/drawer/ModalNavigationDrawer.kt

```diff
@@ -0,0 +1,48 @@
+package app.k9mail.core.ui.compose.designsystem.organism.drawer
+
+import androidx.compose.material3.DrawerValue
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import androidx.compose.material3.ModalNavigationDrawer as Material3ModalNavigationDrawer
+
+@Composable
+fun ModalNavigationDrawer(
+    drawerContent: @Composable (closeDrawer: () -> Unit) -> Unit,
+    modifier: Modifier = Modifier,
+    gesturesEnabled: Boolean = true,
+    content: @Composable (openDrawer: () -> Unit) -> Unit,
+) {
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+    val scope = rememberCoroutineScope()
+
+    Material3ModalNavigationDrawer(
+        drawerContent = {
+            drawerContent {
+                scope.launch {
+                    delay(DRAWER_CLOSE_DELAY)
+                    drawerState.close()
+                }
+            }
+        },
+        modifier = modifier,
+        drawerState = drawerState,
+        gesturesEnabled = gesturesEnabled,
+        content = {
+            content {
+                scope.launch {
+                    drawerState.open()
+                }
```

Comment:
Does this need to launch a coroutine? We could probably call `drawerState.open()` directly.

Reply:
The `DrawerState.open()` and  `DrawerState.close()` functions are suspending as they internally trigger an animation.

Reply:
The Material 3 API got better as with Material 2, but it is still not perfect.